### PR TITLE
Record creationTime of partitions

### PR DESCRIPTION
**Pull request by anwar6953**
https://issues.cask.co/browse/CDAP-2746
http://builds.cask.co/browse/CDAP-DUT2080-5

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionDetail.java [hunk 1]
```diff
@@ -25,4 +25,9 @@
    * Gets the metadata for the partition
    */
   PartitionMetadata getMetadata();
+
+  /**
+   * Gets the creation time of the partition, in milliseconds
+   */
+  long getCreationTime();
```
Comment (anew) [inline cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionDetail.java]: Shouldn't it be part of the PartitionMetadata?

Comment (anwar6953) [inline cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionDetail.java]: Updated; moved creationTime into `PartitionMetadata`

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java [hunk 2]
```diff
@@ -71,6 +71,8 @@
   protected static final byte[] RELATIVE_PATH = { 'p' };
   protected static final byte[] FIELD_PREFIX = { 'f', '.' };
   protected static final byte[] METADATA_PREFIX = { 'm', '.' };
+  // also used as the rowkey prefix of the creation time index
+  protected static final byte[] CREATION_TIME = { 'c', '.' };
```
Comment (anew) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java]: why the trailing dot?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java [hunk 3]
```diff
@@ -315,6 +322,7 @@ protected void getPartitions(@Nullable PartitionFilter filter, PartitionConsumer
         }
         byte[] pathBytes = row.get(RELATIVE_PATH);
         if (pathBytes != null) {
+          byte[] creationTime = row.get(CREATION_TIME);
```
Comment (anew) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java]: this does not seem to be used...

Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java]: right, the time is now being read in `metadataFromRow` method. Removed this unnecessary

**Comment by anew** [comment]
Only two comments from me. Otherwise LGTM.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java (hunk 2 above)
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java]: I was originally having another table, as the index on the creation time, but removed that from this PR.
So, removed the trailing dot, as its not necessary.

**Comment by anwar6953** [comment]
Updated.

**Comment by anew** [comment]
I assume the last commit contains all the previous ones and no further changes except the two small changes suggested in the last round. LGTM

**Pull request merged**